### ISSUE TITLE
[DisallowConcurrentExecution] causing job to only execute once

### ISSUE THREAD

**Issue by jynxeh** [quartznet/quartznet#854]
### Expected behavior
Adding [DisallowConcurrentExecution] to job class is expected to prevent multiple instances running at the same time.

### Actual behavior
After [DisallowConcurrentExecution] attribute has been added the job executes once successfully but does not execute again. Stopping and restarting the scheduler will cause the job to execute once more, but again no more times. Removing the attribute and the job repeats every 10 seconds as expected.

### Steps to reproduce
I have been pairing back the job to try to identify the issue and as part of the investigation changed the execute to log currently executing jobs. The simple code is below.

Schedule setup
```csharp
var aiInferenceJob = JobBuilder.Create<AiInference>().WithIdentity("AI Inference").RequestRecovery(false).Build();
var aiInferenceJobTrigger = TriggerBuilder.Create().WithIdentity("AIInference")
.StartAt(DateTime.Now)
.ForJob(aiInferenceJob)
.WithSimpleSchedule(s => s.WithIntervalInSeconds(10).RepeatForever()).Build();
scheduler.ScheduleJob(aiInferenceJob, aiInferenceJobTrigger);
```
AiInference Job
```csharp
    public class AiInference : IJob
    {
        public void Execute(IJobExecutionContext context)
        {
            Console.WriteLine($"{nameof(AiInference)} started.");
            var currentJobs = context.Scheduler.GetCurrentlyExecutingJobs();

            foreach (var j in currentJobs)
            {
                Console.WriteLine($"Current job: {j.JobDetail.Key.Name}");
            }

            Console.WriteLine($"{nameof(AiInference)} finished.");
        }
    }
```

While investigating the cause I removed the [DisallowConcurrentExecution] attribute and when I do I see the following output:
```
AiInference started.
Current job: AI Inference
AiInference finished.
AiInference started.
Current job: AI Inference
Current job: AI Inference
AiInference finished.
AiInference started.
Current job: AI Inference
Current job: AI Inference
Current job: AI Inference
AiInference finished.
```

Am I correct in thinking that this indicates the scheduler is seeing multiple instances of the job running and therefore does this explain why after the first execution, when the attribute is present, I don't get further triggers? If so, what could be causing this?

I appreciate this is unlikely to be a framework issue but any help is appreciated, thanks.

**Comment by lahma** [comment]
One thing that strikes odd is the `.StartAt(DateTime.Now)`, don't know when it's going to happen as Quartz is expecting UTC times, should be `.StartAt(DateTime.UtcNow)`. It's hard to say what's happening, but I'd suggest enabling logging on DEBUG/INFO level for Quartz to get some insights. If you are running RAMJobStore the scheduling can always happen like in your code, but with persistent store should check whether data already exists.

**Comment by jynxeh** [comment]
@lahma thanks for your comments, I found the cause of this within minutes of your prompt, it's really appreciated. 

I'd fallen into the old trap of spend too long staring at this and hadn't thought to widen my view and investigate the Quartz logging. As soon as I did I found an exception being thrown by our implementation of the history plugin. Fixing this error resulted in the job firing as expected and the [DisallowConcurrentExecution] attribute worked correctly.

I guess this could be considered an issue, giving that if the history plugin doesn't gracefully handle an error it would prevent a non-concurrent job from executing again, but it's a rock and a hard place really.

I'll leave this issue open for review of this behaviour by someone better placed than myself to make the call.

Thanks to everyone who works on this project.
